tests/components/jellyfin/test_config_flow.py

```diff
@@ -0,0 +1,157 @@
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+URL = "https://example.com"
+USERNAME = "test-username"
+PASSWORD = "test-password"
+
+MOCK_SUCCESFUL_CONNECTION_STATE = {"State": CONNECTION_STATE["ServerSignIn"]}
+MOCK_SUCCESFUL_LOGIN_RESPONSE = {"AccessToken": "Test"}
+MOCK_USER_SETTINGS = {"Id": "123"}
+
+
+async def test_abort_if_existing_entry(hass: HomeAssistant):
+    """Check flow abort when an entry already exist."""
+    MockConfigEntry(domain=DOMAIN).add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "single_instance_allowed"
+
+
+async def test_form(hass: HomeAssistant):
+    """Test the complete configuration form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.jellyfin.config_flow.ConnectionManager.connect_to_address",
+        return_value=MOCK_SUCCESFUL_CONNECTION_STATE,
+    ) as mock_connect, patch(
+        "homeassistant.components.jellyfin.config_flow.ConnectionManager.login",
+        return_value=MOCK_SUCCESFUL_LOGIN_RESPONSE,
+    ) as mock_login, patch(
+        "homeassistant.components.jellyfin.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry, patch(
+        "homeassistant.components.jellyfin.config_flow.API.get_user_settings",
+        return_value=MOCK_USER_SETTINGS,
+    ) as mock_set_id:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_URL: URL,
+                CONF_USERNAME: USERNAME,
+                CONF_PASSWORD: PASSWORD,
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == URL
+    assert result2["data"] == {
+        CONF_URL: URL,
+        CONF_USERNAME: USERNAME,
+        CONF_PASSWORD: PASSWORD,
+    }
+
+    assert len(mock_connect.mock_calls) == 1
+    assert len(mock_login.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+    assert len(mock_set_id.mock_calls) == 1
+
+
+async def test_form_cannot_connect(hass: HomeAssistant):
+    """Test we handle an unreachable server."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    result2 = await hass.config_entries.flow.async_configure(
```

Comment:
I'm missing a patch here to set the expected exception on the library call and to avoid any I/O.